web3/utils/address.py

```diff
@@ -0,0 +1,44 @@
+from eth_typing import (
+    ChecksumAddress,
+    HexAddress,
+)
+from eth_utils import (
+    keccak,
+    to_checksum_address,
+)
+from rlp import (
+    encode,
+)
+
+from web3.types import (
+    HexStr,
+    Nonce,
+)
+
+
+def get_create_address(sender: HexAddress, nonce: Nonce) -> ChecksumAddress:
+    """
+    Determine the resulting `CREATE` opcode contract address for a sender and a nonce.
+    """
+    bytes_sender = bytes.fromhex(sender[2:])
+
+    contract_address = keccak(encode([bytes_sender, nonce])).hex()[-40:]
+    return to_checksum_address(contract_address)
+
+
+def get_create2_address(
+    sender: HexAddress, salt: HexStr, init_code: HexStr
+) -> ChecksumAddress:
+    """
+    Determine the resulting `CREATE2` opcode contract address for a sender, salt and
+    bytecode.
+    """
+    bytes_prefix = bytes.fromhex("0xFF"[2:])
```

Comment:
nit: I could go either way here but `to_bytes` from the `eth-utils` dependency would not only make this bit of code more similar to the rest of the code base (so faster to read / process), but it also would accept both 0x-prefixed and non-0x-prefixed hexes.

Something like: 

```py
contract_address = keccak(
    b"\xff"
    + to_bytes(hexstr=sender)
    + to_bytes(hexstr=salt)
    + keccak(to_bytes(hexstr=init_code))
).hex()[-40:]
```

Thoughts on that?

Reply:
I slightly prefer `bytes.fromhex` because it's built-in, but code readability is king!